csrc/host_ir/executor.cpp

```diff
@@ -273,6 +273,14 @@ void HostIrEvaluator::handle(SetCurrentStream* set_current_stream) {
   setCurrentCUDAStream(getCUDAStream(set_current_stream->stream()));
 }
 
+void HostIrEvaluator::handle(GetCurrentStream* get_current_stream) {
+  auto my_device_index = static_cast<c10::DeviceIndex>(
+      communicator_ ? communicator_->deviceId() : 0);
```

Comment:
Your other PR stores my_device_index_ so you could use it here to save a little typing.